src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/QuicExceptionHelpers.cs

```diff
@@ -15,7 +17,22 @@ internal static void ThrowIfFailed(uint status, string? message = null, Exceptio
 
         internal static Exception CreateExceptionForHResult(uint status, string? message = null, Exception? innerException = null)
         {
-            return new QuicException($"{message} Error Code: {MsQuicStatusCodes.GetError(status)}", innerException);
+            return new QuicException($"{message} Error Code: {MsQuicStatusCodes.GetError(status)}", innerException, MapMsQuicStatusToHResult(status));
+        }
+
+        internal static int MapMsQuicStatusToHResult(uint status)
```

Comment:
I assume this is a temporary solution and we should define our own status codes that'll abstract the msquic ones.
That could be probably covered with #32066.

Reply:
We may change the exception base and define quic specific codes. #32066 is primarily talking about the messages.

However, the HResult I pick are generally portable and for example you can use https://www.hresult.info to look them up. What comes out should pretty match what is happening at MsQuic. 
Hopefully this will not be thrown directly in 6.0 and we can finalize this when we do exception cleanup and when we make it public.